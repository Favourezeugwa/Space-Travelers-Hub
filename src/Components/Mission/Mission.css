.mission-btn {
  color: rgb(255, 106, 0);
}
