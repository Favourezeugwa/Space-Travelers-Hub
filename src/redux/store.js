import { configureStore } from '@reduxjs/toolkit';
import rocketReducer from './rocket/rocketSlice';

export default configureStore({
  reducer: {
    rocket: rocketReducer,
  },
});
