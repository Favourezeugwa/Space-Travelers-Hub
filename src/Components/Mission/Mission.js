import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Table } from 'react-bootstrap';
import { getMissionsData } from '../../redux/mission/missionSlice';
import './Mission.css';

function Mission() {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getMissionsData());
  }, [dispatch, getMissionsData]);
  const missions = useSelector((state) => state.mission);
  console.log(missions);
  return (
    <Table striped bordered hover size="lg" className="container my-5">
      <thead>
        <tr>
          <th>Mission</th>
          <th>Description</th>
          <th>Status</th>
        </tr>
      </thead>
      <tbody>
        {missions.map((mission) => (
          <tr key={mission.id}>
            <td>{mission.name}</td>
            <td>{mission.description}</td>
            <div className="btns">
              <tr>
                <button type="button" className="btn btn-secondary ms-3">
                  NOT A MEMBER
                </button>
                <button type="button" className="btn btn-outline-dark ms-3">
                  Join Mission
                </button>
              </tr>
            </div>
          </tr>
        ))}
        {/* <tr>
          <td>Jacob</td>
          <td>Thornton</td>
          <div className="btns">
            <tr>
              <button type="button" className="btn btn-info ms-3">
                NOT A MEMBER
              </button>
              <button type="button" className="btn btn-outline-danger ms-3">
                Join Mission
              </button>
            </tr>
          </div>
        </tr> */}
      </tbody>
    </Table>
  );
}

export default Mission;
