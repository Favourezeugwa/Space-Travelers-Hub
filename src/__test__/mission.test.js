import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Mission from '../Components/Mission/Mission';
import store from '../redux/store';

it('Test Mission renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Mission />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
