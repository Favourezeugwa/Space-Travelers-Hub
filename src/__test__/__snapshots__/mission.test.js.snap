// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Mission renders correctly 1`] = `
<div
  className="container"
>
  <table
    className="container my-5 table table-lg table-striped table-bordered table-hover"
  >
    <thead>
      <tr>
        <th>
          Mission
        </th>
        <th>
          Description
        </th>
        <th>
          Status
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
